### ISSUE TITLE
[tex] $$\begin{split}...\end{split}$$ not numbered with tags: 'all'

### ISSUE THREAD

**Issue by dpvc** [mathjax/MathJax#2272]
When `tags: 'all'` is set (in v3) or `autoNumber: 'all'` (in v2), every displayed equation should have a number, but

```
$$
\begin{split}
...
\end{split}
$$
```

doesn't get numbered.  See this [stackexchange post](https://stackoverflow.com/questions/59141529/mathjax-equation-numbers-do-not-show-using-jekyll-on-github-pages/)

**Comment by dpvc** [comment]
I am not able to reproduce in v3, now, so I think I probably tested for it incorrectly.  It seems to be only a v2 problem.

**Comment by StatProofBook** [comment]
How can one test this fix?

**Comment by dpvc** [comment]
> How can one test this fix?

You could load MathJax from

```
<script src="https://cdn.jsdelivr.net/gh/dpvc/MathJax@issue2272/unpacked/MathJax.js?config=TeX-AMS_CHTML"></script>
```

for testing purposes.  This will use the branch from my fork of the repo that contains the fix rather than the release version.  This branch now contains both the fix for tagging the split environment, and the one for labeling it.

**Comment by StatProofBook** [comment]
Thank you. I have tested three examples and can confirm that it works: `split` environments now receive an equation number and this equation number can be referenced, regardless of whether the environment is enclosed in an `equation` environment or not. The full syntax in my case was `$$ \label{eq:XYZ} \begin{split} ... \end{split} $$` and `\eqref{eq:XYZ}`.

**Comment by StatProofBook** [comment]
@dpvc: Could you notify us in this thread, once this is official MathJax software?

**Comment by dpvc** [comment]
Thanks for the confirmation.  Glad it is working for you.

**Comment by dpvc** [comment]
> Could you notify us in this thread, once this is official MathJax software?

Announcements of new versions are made on the [MathJax User's Forum](https://groups.google.com/forum/?fromgroups#!forum/mathjax-users) and the [MathJax Developer's Forum](https://groups.google.com/forum/?fromgroups#!forum/mathjax-dev) (which has less traffic).  There will also likely be some activity on this issue when the release is drawing near (in particular, when test cases are added for the issue, which usually happens as I am testing the final release candidate).

**Comment by StatProofBook** [comment]
Thanks! We re-check frequently.

**Comment by StatProofBook** [comment]
@dpvc: Because the CDN link

```https://cdn.jsdelivr.net/gh/dpvc/MathJax@issue2272/unpacked/MathJax.js?config=TeX-AMS_CHTML```

was not working anymore, I replaced it by

```https://cdn.jsdelivr.net/gh/mathjax/MathJax@legacy-v2-develop/unpacked/MathJax.js?config=TeX-AMS_CHTML```

Would there be a recommendation to switch to Version 3 and if yes, how would I implement that? (There doesn't some to be a "MathJax.js" in V3 anymore...)

**Comment by dpvc** [comment]
The URL I gave you was only intended for testing purposes, not general use.  I have been preparing a v2.7.8 release, and merged the `issue2272` branch and deleted the branch, so that is why it stopped working for you.  I would **not** use the `legacy-v2-develop` branch in production.  You could use

```
https://cdn.jsdelivr.net/gh/dpvc/MathJax@504a9b7/unpacked/MathJax.js
```

which references the original commit that was in the `issue2272` branch instead for now.

> Would there be a recommendation to switch to Version 3

That is up to you.  Not all the v2 features are available in v3 yet (e.g., line breaking), so you may not be ready to make that move yet.  on the other hand, it is faster, so that might be reason to do so.

See the [MathJax documentation](http://docs.mathjax.org/en/latest/web/start.html) for how to load MathJax version 3.  See the [Upgrading from v2](http://docs.mathjax.org/en/latest/upgrading/v2.html) document for details about the differences between the versions.  If you have a configuration block in your document, you will need to change how you configure MathJax for v3, but there is a [conversion tool](https://mathjax.github.io/MathJax-demos-web/convert-configuration/convert-configuration.html) to help make that easier.  If you are calling the MathJax API (e.g., `MathJax.Hub.Typeset()`), that code will also have to be changed to use the new API.  See the section on [typesetting and converting math](http://docs.mathjax.org/en/latest/web/typeset.html), and the [v2 compatibility examples](http://docs.mathjax.org/en/latest/upgrading/v2.html#v2-compatibility-example) for some information on that.

**Comment by StatProofBook** [comment]
Thanks for the extensive documentation, that is really helpful!

**Comment by dpvc** [comment]
In testsuite:

LaTeXToMathML/environments/issue2272.html

**Comment by sboukortt** [comment]
I’m not sure how related this is but it seems that when numbering equations manually, if they contain a `split` environment, then the `\tag` must come after it.

That is, this doesn’t work:

```latex
\begin{equation}
  \tag{15}
  \begin{split}
    x = y + z \\
          - y \cdot z
  \end{split}
\end{equation}
```

But this does:

```latex
\begin{equation}
  \begin{split}
    x = y + z \\
          - y \cdot z
  \end{split}
  \tag{15}
\end{equation}
```

**Comment by dpvc** [comment]
@sboukortt, this was fixed in v2.7.8 so that both your expressions work.  If they don't work for you, then you are likely using an earlier version, so might want to update to a more recent version.  The highest v2 release is v2.7.9, and v3.2.2 is the current version, with v4.0.0 out in beta release.

**Comment by sboukortt** [comment]
Oops 😁 Thank you. I’m using MathJax through a Hugo theme that is overall maintained, so I think I automatically assumed that it would be using a recent version, but apparently not. I’ll file an issue there.

Thanks again.
